boat-scaffold/src/main/java/com/backbase/oss/codegen/angular/BoatAngularGenerator.java

```diff
@@ -0,0 +1,831 @@
+            additionalProperties.put("tsVersion", ">=3.9.2 <4.0.0");
+        } else if (ngVersion.atLeast("9.0.0")) {
+            additionalProperties.put("tsVersion", ">=3.6.0 <3.8.0");
+        } else if (ngVersion.atLeast("8.0.0")) {
+            additionalProperties.put("tsVersion", ">=3.4.0 <3.6.0");
+        } else if (ngVersion.atLeast("7.0.0")) {
+            additionalProperties.put("tsVersion", ">=3.1.1 <3.2.0");
+        } else {
+            // Angular v6 requires typescript ">=2.7.2 and <2.10.0"
+            additionalProperties.put("tsVersion", ">=2.7.2 and <2.10.0");
+        }
+
+        // Set the rxJS version compatible to the Angular version
+        if (ngVersion.atLeast("10.0.0")) {
+            additionalProperties.put("rxjsVersion", "6.6.0");
+        } else if (ngVersion.atLeast("9.0.0")) {
+            additionalProperties.put("rxjsVersion", "6.5.3");
+        } else if (ngVersion.atLeast("8.0.0")) {
+            additionalProperties.put("rxjsVersion", "6.5.0");
+        } else if (ngVersion.atLeast("7.0.0")) {
+            additionalProperties.put("rxjsVersion", "6.3.0");
+        } else {
+            // Angular v6
+            additionalProperties.put("rxjsVersion", "6.1.0");
+        }
+
+        // Specific ng-packagr configuration
+        if (ngVersion.atLeast("10.0.0")) {
+            additionalProperties.put("ngPackagrVersion", "10.0.3");
+            additionalProperties.put("tsickleVersion", "0.39.1");
+        } else if (ngVersion.atLeast("9.0.0")) {
+            additionalProperties.put("ngPackagrVersion", "9.0.1");
+            additionalProperties.put("tsickleVersion", "0.38.0");
+        } else if (ngVersion.atLeast("8.0.0")) {
+            additionalProperties.put("ngPackagrVersion", "5.4.0");
+            additionalProperties.put("tsickleVersion", "0.35.0");
+        } else if (ngVersion.atLeast("7.0.0")) {
+            // compatible versions with typescript version
+            additionalProperties.put("ngPackagrVersion", "5.1.0");
+            additionalProperties.put("tsickleVersion", "0.34.0");
+        } else {
+            // angular v6
+            // compatible versions with typescript version
+            additionalProperties.put("ngPackagrVersion", "3.0.6");
+            additionalProperties.put("tsickleVersion", "0.32.1");
+        }
+
+        // set zone.js version
+        if (ngVersion.atLeast("9.0.0")) {
+            additionalProperties.put("zonejsVersion", "0.10.2");
+        } else if (ngVersion.atLeast("8.0.0")) {
+            additionalProperties.put("zonejsVersion", "0.9.1");
+        } else {
+            // compatible versions to Angular 6+
+            additionalProperties.put("zonejsVersion", "0.8.26");
+        }
+
+        //Files for building our lib
+        supportingFiles.add(new SupportingFile("package.mustache", getIndexDirectory(), "package.json"));
+        supportingFiles.add(new SupportingFile("tsconfig.mustache", getIndexDirectory(), "tsconfig.json"));
+    }
```

Comment:
This entire section seems quite long. Can't we call super on this one?

Reply:
Those configs are not present on the Abstract Typescript one, I believe...